lib/common/client/middleware/response/raise_error.rb

```diff
@@ -4,9 +4,49 @@ module Client
     module Middleware
       module Response
         class RaiseError < Faraday::Response::Middleware
+          attr_reader :service_acronym, :error_klass, :detail_key, :code_key, :source_key, :meta_key
+
+          def initialize(app, service_acronym, options = {})
+            @service_acronym = service_acronym
+            @error_klass = Common::Exceptions::BackendServiceException
+            @detail_key  = options[:detail_key]  || 'detail'
+            @code_key    = options[:code_key]    || 'code'
+            @source_key  = options[:source_key]  || 'source'
+            @meta_key    = options[:meta_key]    || 'meta'
+
+            super(app)
+          end
+
           def on_complete(env)
             return if env.success?
-            raise Common::Client::Errors::BackendServiceError.new(env.status.to_i, env[:body])
+            raise error_klass.new(i18n_error_key(env), response_values(env))
+          end
+
+          private
+
+          def code(env)
+            "#{service_acronym.upcase}#{env[:body][code_key]}"
+          end
+
+          def i18n_error_key(env)
+            if I18n.exists?("common.exceptions.#{code(env)}")
+              code(env)
+            else
+              Rails.logger.warn "The following exception should be added to locales: \n #{response_values(env)}"
+              'backend_service_exception'
```

Comment:
It looks like you can pass it an exception without raising it, so I think you could do something like this:

```
exc = UnhandledBackendServiceExceptionException(blah, blah)
Raven.capture_exception(exc)
```